mesonpy/_dylib.py

```diff
@@ -27,20 +28,28 @@ def _otool(self, *args: str) -> str:
     def _install_name_tool(self, *args: str) -> str:
         return subprocess.check_output(['install_name_tool', *args, self._path], stderr=subprocess.STDOUT).decode()
 
+    def _find_tag(self, name: str, field: str) -> List[str]:
+        entries = []
+        # Run otool -l to get the load commands
+        otool_output = self._otool('-l').strip()
+        # Manually parse the output and find the tag entries
+        looking_at_tag = False
+        for line in map(str.split, otool_output.splitlines()):
+            if line[0] == 'cmd':
+                if len(line) < 2:
+                    raise ValueError('Unexpected "otool -l" output')
```

Comment:
I'm not sure being too picky about the exact format of the `otool` output is the right thing: it is not something we control and I don't know what are the backward compatibility guarantees for it, but I suspect not that many. I would write this to be as flexible as possible and to do not error out on unexpected input.

Reply:
I disagree, IMO it's better to error out instead of keeping working with unexpected output, possibly causing obscure errors that are hard to debug. If this fails, we reproduce it, adjust the logic to expect some new format, and make a new (bugfix if needed) release.